src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -321,7 +322,7 @@ def extract_archive_argv(
     ) -> tuple[str, ...]:
         # Note that the `output_dir` must already exist.
         # The caller should validate that it's a valid `.tar` file.
-        prog_args = ("-Ilz4",) if archive_suffix == ".tar.lz4" else ()
+        prog_args = ("-Ilz4",) if archive_suffix == ".tar.lz4" and sys.platform == "linux" else ()
```

Comment:
Using `sys.platform` is not safe because it uses localhost, even when using Docker and Remote Execution environments. You should request `Platform` inside the rule instead and match the enum.

Reply:
There is no rule here, this is pure utils, hm...